packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol

```diff
@@ -325,15 +325,15 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone, ISemver {
         } else if (_ident == LocalPreimageKey.DISPUTED_OUTPUT_ROOT) {
             // Load the disputed proposal's output root
             oracle.loadLocalData(_ident, uuid.raw(), disputed.raw(), 32, _partOffset);
-        } else if (_ident == LocalPreimageKey.STARTING_L2_BLOCK_NUMBER) {
-            // Load the starting proposal's L2 block number as a big-endian uint64 in the
+        } else if (_ident == LocalPreimageKey.DISPUTED_L2_BLOCK_NUMBER) {
+            // Load the disputed proposal's L2 block number as a big-endian uint64 in the
             // high order 8 bytes of the word.
 
-            // If the starting position is 0 (invalid), the starting output root is genesis. Otherwise,
-            // we add the index at depth + 1 to the genesis block number to get the L2 block number.
+            // If the starting position is 0 (invalid), the disputed output root is at genesis + 1. Otherwise,
+            // we add the index at depth + 2 to the genesis block number to get the disputed L2 block number.
             uint256 l2Number = startingPos.raw() == 0
-                ? GENESIS_BLOCK_NUMBER
-                : GENESIS_BLOCK_NUMBER + startingPos.traceIndex(SPLIT_DEPTH) + 1;
+                ? GENESIS_BLOCK_NUMBER + 1
+                : GENESIS_BLOCK_NUMBER + startingPos.traceIndex(SPLIT_DEPTH) + 2;
```

Comment:
It may make more sense to use `disputedPos + 1` instead of `startingPos + 2` here.

Reply:
Actually this is definitely incorrect, it should just be:
```
uint256 l2Number = GENESIS_BLOCK_NUMBER + disputedPos.traceIndex(SPLIT_DEPTH) + 1;
```

A trace index of 0 is valid - it's the first block after genesis. The absolute prestate is effectively trace index -1.